### PR TITLE
properly disconnect from relays

### PR DESCRIPTION
**Pull request by benthecarman**
eventually this would cause resource exhaustion

### CODE REVIEW

File: mutiny-core/src/lib.rs [hunk 1]
```diff
@@ -235,6 +235,10 @@ impl<S: MutinyStorage> MutinyWallet<S> {
                         }
                     }
                 }
+
```
Comment (TonyGiorgio) [inline mutiny-core/src/lib.rs:238]: shouldn't the `read_fut` above handle `Err` so it breaks out of the loop and disconnects?

Comment (benthecarman) [reply mutiny-core/src/lib.rs:238]: Don't see what you're talking about, looks like every error is handled, the disconnect is outside the inner `loop`

Comment (TonyGiorgio) [reply mutiny-core/src/lib.rs:238]: when read errors happen it just keeps going inside the inner loop.

Comment (benthecarman) [reply mutiny-core/src/lib.rs:243]: Ah I see, good call out

**Review by TonyGiorgio — APPROVED** [review]

**Pull request merged**